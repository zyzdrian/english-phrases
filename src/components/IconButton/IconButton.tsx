import React from 'react';
import { Button, IButton } from '..';
import leftArrow from './left-arrow.svg';
import styled from 'styled-components';

const StyledIconButton = styled(Button)`
max-width: 56px;
display: flex;
justify-content: center;
`;

export const IconButton: React.FC<IButton> = (props) => (
    <StyledIconButton {...props}>
        <img src={leftArrow} alt='<-' />
    </StyledIconButton>
);
