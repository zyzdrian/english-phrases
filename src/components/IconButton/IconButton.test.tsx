import React from 'react';
import { IconButton } from './IconButton';
import { render, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { renderTheme, lightTheme } from '../../styledComponents';

describe('<IconButton /> tests', () => {
    it('renders without props', () => {
        const { container, getByRole } = renderTheme(<IconButton />);

        expect(container.firstChild).toMatchSnapshot();
        expect(getByRole('button')).toHaveTextContent('');
        expect(getByRole('button')).not.toHaveAttribute('disabled');
        expect(getByRole('button')).toHaveStyle(`background-color: ${lightTheme.primary}`);
    });

    it('renders with prop disabled', () => {
        const { container, getByRole } = render(<IconButton disabled />);

        expect(container.firstChild).toMatchSnapshot();
        expect(getByRole('button')).toHaveAttribute('disabled');
    });

    it('renders as secondary', () => {
        const { container, getByRole } = renderTheme(<IconButton variant='secondary' />);

        expect(container.firstChild).toMatchSnapshot();
        expect(getByRole('button')).toHaveStyle(`background-color: ${lightTheme.secondary}`);
    });

    it('renders with prop disabled', () => {
        const onClick = jest.fn();

        const { container, getByRole } = render(<IconButton onClick={onClick} />);
        fireEvent.click(getByRole('button'));

        expect(container.firstChild).toMatchSnapshot();
        expect(onClick).toBeCalled();
    });
});
