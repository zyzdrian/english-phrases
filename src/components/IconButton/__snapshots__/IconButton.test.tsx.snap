// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IconButton /> tests renders as secondary 1`] = `
<button
  class="sc-bwzfXH dPTGrP sc-cSHVUG VRnhl"
>
  <img
    alt="<-"
    src="left-arrow.svg"
  />
</button>
`;

exports[`<IconButton /> tests renders with prop disabled 1`] = `
<button
  class="sc-bwzfXH gtkPbW sc-cSHVUG VRnhl"
  disabled=""
>
  <img
    alt="<-"
    src="left-arrow.svg"
  />
</button>
`;

exports[`<IconButton /> tests renders with prop disabled 2`] = `
<button
  class="sc-bwzfXH gtkPbW sc-cSHVUG VRnhl"
>
  <img
    alt="<-"
    src="left-arrow.svg"
  />
</button>
`;

exports[`<IconButton /> tests renders without props 1`] = `
<button
  class="sc-bwzfXH cUmWVg sc-cSHVUG VRnhl"
>
  <img
    alt="<-"
    src="left-arrow.svg"
  />
</button>
`;
